extensions/resource-deployment/src/ui/deployAzureSQLDBWizard/deployAzureSQLDBWizardModel.ts

```diff
@@ -29,9 +42,149 @@ export class DeployAzureSQLDBWizardModel extends Model {
 	public databaseCollation!: string;
 	public newFirewallRule!: boolean;
 
+	public get notebookService(): INotebookService {
+		return this.wizard.notebookService;
+	}
+
+	public get toolService(): IToolsService {
+		return this.wizard.toolsService;
+	}
+
+	constructor(public sqldbProvider: AzureSQLDBDeploymentProvider, wizard: ResourceTypeWizard) {
+		super(sqldbProvider, wizard);
+		this.wizard.wizardObject.title = constants.WizardTitle;
+	}
+
+	initialize(): void {
+		this.wizard.setPages(this.getPages());
+		this.wizard.wizardObject.generateScriptButton.hidden = true;
+		this.wizard.wizardObject.doneButton.label = constants.WizardDoneButtonLabel;
+	}
+
+	async onOk(): Promise<void> {
```

Comment:
There is a lot of code repeated in individual wizard specific models. Can we move it to ResourceModel?